crates/auth/src/auth/permissions/validator.rs

```diff
@@ -0,0 +1,598 @@
+                        }))],
+                        _ => vec![],
+                    }
+                } else {
+                    vec![]
+                }
+            },
+        },
+        PathMapping {
+            pattern: "^/contexts/([^/]+)/alias/([^/]+)$",
+            handler: |components, method| {
+                match (components.get(1), components.get(2)) {
+                    (Some(&ctx_id), Some(&user_id)) => vec![Permission::Context(ContextPermission::Alias(AliasPermission::Lookup {
+                        context_id: Some(ctx_id.to_string()),
+                        user_id: Some(user_id.to_string()),
+                    }))],
+                    _ => vec![],
+                }
+            },
+        },
+        // Blob Operations
+        PathMapping {
+            pattern: "^/blobs/stream$",
+            handler: |_, method| match method {
+                HttpMethod::POST => vec![Permission::Blob(BlobPermission::Add(AddBlobPermission::Stream))],
+                _ => vec![],
+            },
+        },
+        PathMapping {
+            pattern: "^/blobs/file$",
+            handler: |_, method| match method {
+                HttpMethod::POST => vec![Permission::Blob(BlobPermission::Add(AddBlobPermission::File))],
+                _ => vec![],
+            },
+        },
+        PathMapping {
+            pattern: "^/blobs/url$",
+            handler: |_, method| match method {
+                HttpMethod::POST => vec![Permission::Blob(BlobPermission::Add(AddBlobPermission::Url))],
+                _ => vec![],
+            },
+        },
+        PathMapping {
+            pattern: "^/blobs/([^/]+)$",
+            handler: |components, method| {
+                if let Some(&blob_id) = components.get(1) {
+                    match method {
+                        HttpMethod::DELETE => vec![Permission::Blob(BlobPermission::Remove(
+                            ResourceScope::Specific(vec![blob_id.to_string()])
+                        ))],
+                        _ => vec![],
+                    }
+                } else {
+                    vec![]
+                }
+            },
+        },
+    ];
+}
+
+impl PermissionValidator {
+    pub fn new() -> Self {
+        Self
+    }
+
+    /// Determine required permissions for a given request
+    pub fn determine_required_permissions(&self, request: &Request<Body>) -> Vec<Permission> {
+        let path = request.uri().path();
+        let method = match request.method().as_str() {
+            "GET" => HttpMethod::GET,
+            "POST" => HttpMethod::POST,
+            "PUT" => HttpMethod::PUT,
+            "DELETE" => HttpMethod::DELETE,
+            "PATCH" => HttpMethod::PATCH,
+            _ => HttpMethod::Any,
+        };
+
+        let mut required_permissions = Vec::new();
+
+        // First check path mappings
+        for mapping in PATH_MAPPINGS.iter() {
+            if let Ok(regex) = Regex::new(mapping.pattern) {
```

Comment:
since the regexes are static consider pre-compiling them on auth server startup, instead of on every request, you can call `Regex::new` in a `lazy_static`, and if you only want to compile on demand and reuse, use [`LazyLock`](https://doc.rust-lang.org/std/sync/struct.LazyLock.html)